bot/slack/commands/ip_info.py

```diff
@@ -0,0 +1,49 @@
+import netaddr
+import os
+
+from mozdef_util.geo_ip import GeoIP
+
+
+def is_ip(ip):
+    try:
+        netaddr.IPNetwork(ip)
+        return True
+    except Exception:
+        return False
+
+
+def ipLocation(ip):
+    location = ""
+    try:
+        geoip_data_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "../../../data/GeoLite2-City.mmdb")
+        geoip = GeoIP(geoip_data_dir)
+        geoDict = geoip.lookup_ip(ip)
+        if geoDict is not None:
+            if 'error' in geoDict:
+                return geoDict['error']
+            location = geoDict['country_name']
+            if geoDict['country_code'] in ('US'):
+                if geoDict['metro_code']:
+                    location = location + '/{0}'.format(geoDict['metro_code'])
+    except Exception:
+        location = ""
+    return location
+
+
+class command():
+    def __init__(self):
+        self.command_name = '!ipinfo'
+        self.help_text = 'Perform a geoip lookup on an ip address'
+
+    def handle_command(self, parameters):
+        response = ""
+        for ip_token in parameters:
+            if is_ip(ip_token):
+                ip = netaddr.IPNetwork(ip_token)[0]
+                if (not ip.is_loopback() and not ip.is_private() and not ip.is_reserved()):
```

Comment:
Is it not considered bad style to put conditions in parens when they don't have to be?
This condition _could_ be shortened from

```py
if not A and not B and not C
```

to

```py
if not (A or B or C)
```

as per [DeMorgan's Laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws).

Reply:
Fixed in https://github.com/mozilla/MozDef/pull/1018/commits/f0f30519a71750afa8dea1db5de8f67f622be039